File: modules/common/src/main/java/com/sygic/modules/common/di/component/module/PoiDataManagerModule.java
package com.sygic.modules.common.di.component.module;

import com.sygic.modules.common.poi.manager.PoiDataManager;
import com.sygic.modules.common.poi.manager.PoiDataManagerImpl;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class PoiDataManagerModule {

    @Singleton
    @Provides
    PoiDataManager providePoiDataManager() {
        return new PoiDataManagerImpl();
    }
}

File: modules/common/src/main/java/com/sygic/modules/common/di/PoiDataManagerModule.kt
package com.sygic.modules.common.di

import com.sygic.modules.common.poi.manager.PoiDataManager
import com.sygic.modules.common.poi.manager.PoiDataManagerImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PoiDataManagerModule {

    @Singleton
    @Provides
    fun providePoiDataManager(): PoiDataManager = PoiDataManagerImpl()
}